wodles/aws/aws_s3.py

```diff
@@ -732,9 +738,44 @@ def iter_regions_and_accounts(self, account_id, regions):
         raise NotImplementedError
 
     def iter_events(self, event_list, log_key, aws_account_id):
+        def check_recursive(json_item=None, nested_field: str = '', regex: str = ''):
+            field_list = nested_field.split('.', 1)
+            try:
+                expression_to_evaluate = json_item[field_list[0]]
+            except TypeError:
+                if isinstance(json_item, list):
+                    for i in json_item:
+                        if check_recursive(i, field_list[0], regex=regex):
+                            return True
+                return False
+            except KeyError:
+                return False
+            if len(field_list) == 1:
+                def check_regex(exp):
+                    try:
+                        return re.match(regex, exp) is not None
+                    except TypeError:
+                        if isinstance(exp, list):
+                            for ex in exp:
+                                if check_regex(ex):
+                                    return True
+                    return False
+
+                return check_regex(expression_to_evaluate)
+            return check_recursive(expression_to_evaluate, field_list[1], regex=regex)
+
+        def event_should_be_skipped():
+            if self.discard_field and self.discard_regex:
+                if check_recursive(event, nested_field=self.discard_field, regex=self.discard_regex):
+                    return True
+            return False
+
         if event_list is not None:
             for event in event_list:
-                # Parse out all the values of 'None'
+                if event_should_be_skipped():
+                    debug(f'+++ The "{self.discard_regex}" regex found a match in the "{self.discard_field}" field. '
+                          f'The event will be skipped.', 1)
```

Comment:
Maybe we can clarify this message a bit by omitting re.compile. For example right now this message could look like this:

```
+++ The "re.compile('.*157441.*')" regex found a match in the "resources.accountId" field. The event will be skipped.
```

How about changing it so that only the pattern appears?Que tal cambiarlo para que solo aparezca el patron:

```
+++ The '.*157441.*' regex found a match in the "resources.accountId" field. The event will be skipped.
```